CONTRIBUTING.rst

```diff
@@ -1,107 +1,414 @@
+|                    |             | - Variables, functions, arguments have clear names |
+|                    |             | - Easy to read, PEP8                               |
+|   `Coding Style`_  |    Any      | - Public functions have docstring (numpydoc format)|
+|                    |             | - Low redundancy                                   |
+|                    |             | - No new dependency                                |
+|                    |             | - Backward compatibility                           |
++--------------------+-------------+----------------------------------------------------+
+|                    |             | - Test type is adapted to function behavior        |
+|                    |             | - Tests pass continuous integration                |
+|                    |  Bugfixes   | - Coverage doesn't decrease                        |
+|      `Tests`_      | New features| - Fast, using small mocked data                    |
+|                    |             | - Atomic (one per function) and seeded             |
+|                    |             | - For Bugfixes: non-regression test                |
++--------------------+-------------+----------------------------------------------------+
+|                    |             | - Clearly showcase benefits                        |
+|      Examples      | New features| - Run in a few seconds                             |
+|                    |             | - Use light data (generated or from Nilearn)       |
+|                    |             | - Renders well after build                         |
++--------------------+-------------+----------------------------------------------------+
+|                    |             | - Simple and didactic                              |
+|  `Documentation`_  |    Any      | - Links to relevant examples                       |
+|                    |             | - Renders well after build                         |
+|                    |             | - Doesn't include code                             |
++--------------------+-------------+----------------------------------------------------+
+
+PR Structure
+-------------
+
+A new pull request must have a clear scope, conveyed through its name, a
+reference to the issue it targets (through the exact mention "Closes #XXXX")
+as well as a synthetic summary of its goals and main steps.
+When working on big contributions, we advise contributors to split them in
+several PRs when possible.
+This has the benefit to make code changes clearer, easier to make and to review
+and overall smoothen the whole process.
+No changes unrelated to the PR should be included.
+
+When relevant, PR names should also include tags if they fall in various
+categories.
+If the authors feels he still has work to do before a first round of review, he
+can include the [WIP] tag or conversely, the [MRG] tag to highlight this PR is
+ready for review.
```

Comment:
We can also encourage users to open their PRs in draft mode and switch to normal mode when they are ready for review. WDYT?

Reply:
That's a good way to put it.

Reply:
I implemented this for now, but from the discussion above, it seemed most people preferred [WIP] and [MRG] tag to that and proposing to use both (as I do for know) seem unclear. WDYT ?

Reply:
I get your point. I don't have a strong opinion on this since both ways are clear enough (to me at least...). I was using [WIP]/[MRG] before but transitioned to Draft/Regular PRs when the feature became available on GitHub. Maybe we can leave the choice to users and we will see which one they prefer in the end? WDYT?